examples/cvode/serial/cvAdvDiff_bnd.c

```diff
@@ -133,24 +129,28 @@ int main(void)
 
   /* Create the SUNDIALS context */
   retval = SUNContext_Create(NULL, &sunctx);
-  if(check_retval(&retval, "SUNContext_Create", 1)) return(1);
+  if(check_retval(retval, "SUNContext_Create")) return(1);
+
+  /* Setup different error handler stack so that we abort after logging */
+  SUNContext_PopErrHandler(sunctx);
+  SUNContext_PushErrHandler(sunctx, SUNAbortErrHandlerFn, NULL);
+  SUNContext_PushErrHandler(sunctx, SUNLogErrHandlerFn, NULL);
 
   /* Get a reference to the profiler */
   retval = SUNContext_GetProfiler(sunctx, &profobj);
-  if(check_retval(&retval, "SUNContext_GetProfiler", 1)) return(1);
+  if(check_retval(retval, "SUNContext_GetProfiler")) return(1);
 
   SUNDIALS_MARK_FUNCTION_BEGIN(profobj);
 
   /* Create a serial vector */
 
   u = N_VNew_Serial(NEQ, sunctx);  /* Allocate u vector */
-  if(check_retval((void*)u, "N_VNew_Serial", 0)) return(1);
+  if(check_retval(SUNGetLastErr(sunctx), "N_VNew_Serial")) return(1);
 
   reltol = ZERO;  /* Set the tolerances */
   abstol = ATOL;
 
   data = (UserData) malloc(sizeof *data);  /* Allocate data memory */
-  if(check_retval((void *)data, "malloc", 2)) return(1);
```

Comment:
I like that the new error handling features allow `check_retval` to be streamlined, but it looks like our example programs can no longer check for internal failures (e.g., `malloc` failure).  Is the assumption that these errors just do not occur, or that if they do occur then the OS should abort the program itself?

Reply:
This is only done in one example for demonstration purposes. We can still leave `check_retval` the same in other examples.

Reply:
What is it demonstrating to leave this out?

Reply:
The point was to streamline `check_retval` in this one example to show an alternate way to check sundials calls. Doing so made `check_retval` not work for the `malloc` case. The `malloc` call is user code, and the user can check it however they want to (or not), so in this example it isn't checked.

Reply:
If I recall correctly, we had a function that we defined in many examples that checked the return value of routines.  We had implicitly encouraged users to use this routine.  Are we no longer doing that?

Reply:
We are doing that, that is what `check_retval` is.

Reply:
Do you have other examples where error checking for both sundials and "user" routines is done?

Reply:
Yes, almost all of our other examples do that and were not changed in this PR.